provider/google.go

```diff
@@ -133,49 +129,33 @@ func NewGoogleProvider(project string, dryRun bool) (Provider, error) {
 }
 
 // Zones returns the list of hosted zones.
-func (p *googleProvider) Zones() (zones []*dns.ManagedZone, _ error) {
+func (p *googleProvider) Zones() (map[string]*dns.ManagedZone, error) {
+	zones := make(map[string]*dns.ManagedZone)
+
 	f := func(resp *dns.ManagedZonesListResponse) error {
-		// each page is processed sequentially, no need for a mutex here.
-		zones = append(zones, resp.ManagedZones...)
+		for _, zone := range resp.ManagedZones {
+			if strings.HasSuffix(zone.DnsName, p.domain) {
+				zones[zone.Name] = zone
+			}
+		}
+
 		return nil
 	}
 
-	err := p.managedZonesClient.List(p.project).Pages(context.TODO(), f)
-	if err != nil {
+	if err := p.managedZonesClient.List(p.project).Pages(context.TODO(), f); err != nil {
 		return nil, err
 	}
 
 	return zones, nil
 }
 
-// CreateZone creates a hosted zone given a name.
-func (p *googleProvider) CreateZone(name, domain string) error {
-	zone := &dns.ManagedZone{
-		Name:        name,
-		DnsName:     domain,
-		Description: "Automatically managed zone by kubernetes.io/external-dns",
-	}
-
-	_, err := p.managedZonesClient.Create(p.project, zone).Do()
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// DeleteZone deletes a hosted zone given a name.
-func (p *googleProvider) DeleteZone(name string) error {
-	err := p.managedZonesClient.Delete(p.project, name).Do()
+// Records returns the list of records in all relevant zones.
+func (p *googleProvider) Records(_ string) (endpoints []*endpoint.Endpoint, _ error) {
+	zones, err := p.Zones()
```

Comment:
I wonder if filtering on Zones here is really necessary, this list is to be used by plan to figure out the list of endpoints to be created, and it might make sense to have ti returns all the records to have a complete picture